library/ssl_tls12_client.c

```diff
@@ -1764,9 +1764,11 @@ static int ssl_parse_server_ecdh_params(mbedtls_ssl_context *ssl,
         return MBEDTLS_ERR_SSL_DECODE_ERROR;
     }
 
+#if !defined(PSA_WANT_ALG_FFDH)
```

Comment:
This is because `ecpoint_len` is of `uint8_t` type, so max value is `255` and when FFDH is enabled `sizeof(handshake->ecdh_psa_peerkey))` is always greater then `255` and we end up with compilator warning.

Even with proposed solution for `DH_PSA_PEERKEY_SIZE` which I think is a good option the problem still exists. If we would go with dynamic allocation then this guard could be removed, but I prefer your solution.

Reply:
Not possible to change the type of `ecpoint_len` to `size_t`?

Reply:
This would hide that the guarded check in configuration with ffdh enabled is redundant. The code optimization is very important in mbed-tls, so I thought we don't want to do that and compile out redundant code instead.

Reply:
> This would hide that the guarded check in configuration with ffdh enabled is redundant.

Why would it be redundant with FFDH enabled?

Reply:
Because `ecpoint_len` is of `uint8_t` so max length is `255` and when FFDH is enabled common `xxdh` buffer is always greater than `255`.

Reply:
The buffer size is always greater than 255 because FFDH key size is greater or equal to 2048 bits which is true and is highly likely to not change but not directly related to ECDH. At least, I would add a comment explaining why we do not need to do the check if FFDH is enabled and probably also a static assertion that if FFDH is enabled then `sizeof(handshake->ecdh_psa_peerkey) >= UINT8_MAX`.

Reply:
Will something like this be ok:

```
    /* When FFDH is enabled the key size is greater or equal to 2048 bits which is greater than
       max key length for any EC group, so this check can be skipped. */
#if !defined(PSA_WANT_ALG_FFDH)
    if (ecpoint_len > sizeof(handshake->ecdh_psa_peerkey)) {
        return MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE;
    }
#else
    MBEDTLS_STATIC_ASSERT(sizeof(handshake->ecdh_psa_peerkey) >= UINT8_MAX, "peer key buffer too small");
#endif
```

Reply:
> When FFDH is enabled the key size is greater or equal to 2048 bits which is greater than
>        max key length for any EC group, so this check can be skipped. */

Rather something like:
"When FFDH is enabled, the array `handshake->xxdh_psa_peer_key` size takes into account the sizes of the FFDH keys which are at least 2048 bits. The size of the array is thus greater than 256 bytes which is greater than any possible value of `ecpoint_len` (type uint8_t) and the check below can be skipped."